### PR TITLE
unpacker.eclass: tests, fixes, .gpkg.tar support

### PR DESCRIPTION
**Pull request by mgorny**
- add tests for unpacking various file formats
- fix handling broken/invalid `.zst` and `.7z` files
- use lowercase suffixes everywhere consistently
- add support for `.lz4` and `.lzo`
- add support for on-the-fly unpacking of the image archive from `.gpkg.tar`
- use parallel xz decompression (by @thesamesam)
- support lbzip2 if available
- fix handling `.deb` that use GNU ar format
- unpack `.deb` on-the-fly (i.e. without temporary files)

Includes an example `.gpkg.tar` unpacking for gentoo-kernel-bin.

CC @gentoo/base-system, @gentoo/dist-kernel

### CODE REVIEW

**Review by thesamesam — APPROVED** [review]

File: eclass/unpacker.eclass [hunk 1]
```diff
@@ -406,7 +406,7 @@ _unpacker() {
 		: ${UNPACKER_LZIP:=$(type -P plzip || type -P pdlzip || type -P lzip)}
 		comp="${UNPACKER_LZIP} -dc" ;;
 	*.zst)
-		comp="zstd -dfc" ;;
+		comp="zstd -dc" ;;
 	esac
```
Comment (thesamesam) [inline eclass/unpacker.eclass]: Add a Closes tag for the bug to this commit

**Comment by mgorny** [comment]
Also updated the tests to compare file contents and print diff on mismatch, plus test multistream compressed files.

**Review by floppym — APPROVED** [review]
Nice work, as usual.

File: eclass/tests/unpacker.sh [hunk 2]
```diff
@@ -0,0 +1,291 @@
+#!/bin/bash
```
Comment (SoapGentoo) [inline eclass/tests/unpacker.sh:1]: portable shebangs?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

**Review by stha09 — APPROVED** [review]

File: eclass/tests/unpacker.sh (hunk 2 above)
Comment (mgorny) [reply eclass/tests/unpacker.sh]: Done.

**Comment by mgorny** [comment]
Updated shebang and fixed grammar per @SoapGentoo.

**Pull request closed without merging**